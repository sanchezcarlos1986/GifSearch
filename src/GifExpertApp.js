import React, {useState} from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

export default function GifExpertApp({defaultCategories = []}) {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div className="GifExpertApp">
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </div>
  );
}
