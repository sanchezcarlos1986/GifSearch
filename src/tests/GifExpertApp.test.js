import React from 'react';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';
import {shallow} from 'enzyme';

describe('GifExpertApp', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {};
    component = shallow(<GifExpertApp {...props} />);
  });

  it('should renders ok', () => {
    expect(component).not.toBeNull();
    expect(component.hasClass('GifExpertApp')).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should shows categories list', () => {
    const categories = ['Dragon Ball', 'Hunter X'];
    component = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(component).toMatchSnapshot();
    expect(component.find('GifGrid').length).toBe(categories.length);
  });
});
