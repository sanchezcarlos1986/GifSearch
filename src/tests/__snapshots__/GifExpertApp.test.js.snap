// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should renders ok 1`] = `
<div
  className="GifExpertApp"
>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</div>
`;

exports[`GifExpertApp should shows categories list 1`] = `
<div
  className="GifExpertApp"
>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="Hunter X"
      key="Hunter X"
    />
  </ul>
</div>
`;
